### DISCUSSION TITLE
How to debug

### DISCUSSION THREAD

**Discussion by karchjd** [alpinejs/alpine#3639] [1. Help]
Hello,

My concrete problem is that my `Lavaan Model` menu is not working anymore. See, https://jsfiddle.net/juka87/dnpztw0m/1/. I create this with `x-for` in lines `134-150` based on a store I create in `288-329`. It used to work until I changed something in the code. However, I am not getting any error messages.

My more general question is how can I debug such things in alpine.js? In a different framework, I would simply put a debugger statement in or before the loop to see what is going on. However, this seems not possible. I am aware of devtools addon but that just seems to expose the state of all component variables. Shouldn't there be an error message?

**Comment by ekwoka** [comment]
An error message can only show if the code causes an error. It can't throw an error just because it doesn't do what you expect.

What does it do now? And what do you expect it to do?

The devtools can be useful, because the state may be different than you thought

The issue here is that that `x-for` is not inside of an `x-data`

Also, add `:key` to for loops.

Also, don't use jquery.

**Comment by karchjd** [comment]
Thanks for your help! That solved the concrete problem. 

Regarding the bigger picture, I realized that my project is too complex for alpine.js. I move on to svelte.js.

**Reply by ekwoka** [reply]
I'd be pretty surprised if that's the case, but yes, AlpineJS is not an alternative to svelte/vue/react.

It's made to add interactivity to an SSR rendered site from outside the JS ecosystem.
